packages/geo/src/lib/catalog/shared/catalog.service.ts

```diff
@@ -809,16 +816,18 @@ export class CatalogService {
           extern = false;
         }
 
-        if (this.testLayerRegexes(layer.id, regexes) === false) {
+        if (
+          this.testLayerRegexes(layer.id.toLocaleString(), regexes) === false
+        ) {
           return undefined;
         }
         const baseSourceOptions = {
```

Comment:
Idéallement on essait autant que possible de forcer l'assignation avec l'opérateur `as`. On pourrait:
```suggestion
        const baseSourceOptions: ArcGISRestDataSourceOptions = {
```

Pour le `type as unknown`, on pourrait créer un type dans le datasource.interface et remplacer le DataSourceOptions.type par ce nouveau DataSourceType et l'utiliser ici pour forcer l'assignation:
```
export type DataSourceType =
  | 'wms'
  | 'wfs'
  | 'vector'
  | 'wmts'
  | 'xyz'
  | 'osm'
  | 'tiledebug'
  | 'carto'
  | 'arcgisrest'
  | 'imagearcgisrest'
  | 'tilearcgisrest'
  | 'websocket'
  | 'mvt'
  | 'cluster';
```

Reply:
J'ai essayé et ca pète de partout. J'ai préféré setter unknown pour cette raison.

Reply:
J'ai fait quelques correctifs en compromis ;)